src/inmanta/agent/handler.py

```diff
@@ -424,20 +425,23 @@ def _log_msg(self, level: int, msg: str, *args: object, exc_info: bool = False,
         if exc_info:
             kwargs["traceback"] = traceback.format_exc()
 
-        for k, v in kwargs.items():
+        def clean_arg_value(k: str, v: object) -> object:
             try:
-                json_encode(v)
+                # make sure we have clean dict
```

Comment:
This comment explains what the code does. What I am missing here is an explanation on why we have to do this.